src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java

```diff
@@ -219,6 +223,42 @@ public AdyenPaymentPluginApi(final AdyenConfigurationHandler adyenConfigurationH
     }
 
     @Override
+    public List<PaymentTransactionInfoPlugin> getPaymentInfo(final UUID kbAccountId, final UUID kbPaymentId, final Iterable<PluginProperty> properties, final TenantContext context) throws PaymentPluginApiException {
+        final List<PaymentTransactionInfoPlugin> transactions = super.getPaymentInfo(kbAccountId, kbPaymentId, properties, context);
+        final ExpiredPaymentPolicy expiredPaymentPolicy = expiredPaymentPolicy(context);
+
+        if (expiredPaymentPolicy.isExpired(transactions)) {
+            cancelExpiredPayment(transactions, context);
+            // reload payment
+            return super.getPaymentInfo(kbAccountId, kbPaymentId, properties, context);
+        }
+        return transactions;
+    }
+
+    private void cancelExpiredPayment(final List<PaymentTransactionInfoPlugin> transactions, final TenantContext context) {
+        final List<PluginProperty> updatedStatusProperties = PluginProperties.buildPluginProperties(
+                ImmutableMap.builder()
+                            .put(PROPERTY_FROM_HPP_TRANSACTION_STATUS,
+                                 PaymentPluginStatus.CANCELED.toString())
+                            .put("message",
+                                 "Payment Expired - Cancelled by Janitor")
+                            .build());
+
+        final PaymentTransactionInfoPlugin expiredPendingTransaction = transactions.get(0);
+        try {
+            dao.updateResponse(expiredPendingTransaction.getKbTransactionPaymentId(),
```

Comment:
I think so, that should also help in the notification handling case.
But I think I'd prefer having a separate PR for that.